### PR TITLE
Update the configure script to use platform agnostic git command, readme with OSX and Arch dependencies and stabilize makefile manpages phony

### PR DESCRIPTION
**Pull request by asynchroza**

### CODE REVIEW

**Comment by mohd-akram** [comment]
Thanks for the PR. Just curious, does `git-restore-mtime` not work for some systems?

File: Makefile [hunk 1]
```diff
@@ -22,4 +22,5 @@ clean:
 $(mansectdir):
 	mkdir -p $@
 
-manpages: $(mansectdir) files.mk $(files)
+manpages: 
+	$(mansectdir) files.mk $(files)
```
Comment (mohd-akram) [inline Makefile]: This isn't right, that's not a command. What's the motivation for this?

Comment (asynchroza) [reply Makefile]: I thought it's part of the build process. I will roll it back.

**Comment by asynchroza** [comment]
> Thanks for the PR. Just curious, does `git-restore-mtime` not work for some systems?

Yes. It seems `git-restore-mtime` is only present on Ubuntu machine (potentially Debian systems). I'm running OSX, and from my research it seems that it's not supported on Arch linux as well.

**Comment by asynchroza** [comment]
Also, it seems that there are plenty of tools in homebrew which overwrite the `MANPATH` since they install their own `man` binary, so the `make install` is a bit unreliable on macos systems.

**Comment by mohd-akram** [comment]
I think you can still run either `git-restore-mtime` or `git restore-mtime` on Homebrew and Arch though. I can see it installs it to `bin` in both cases.

> Also, it seems that there are plenty of tools in homebrew which overwrite the `MANPATH` since they install their own `man` binary, so the `make install` is a bit unreliable on macos systems.

Hmm that's strange. The default for `make install` is in `/usr/local/man` which I think should be picked up by any `man` binary. You can also change it eg. `make install mandir=/some/other/man`. I'm also running macOS but using MacPorts.

**Comment by asynchroza** [comment]
> I think you can still run either git-restore-mtime or git restore-mtime on Homebrew and Arch though. I can see it installs it to bin in both cases.

You can run this `git restore-mtime`, but you first need to install `git-tools` on Homebrew.

> Hmm that's strange. The default for make install is in /usr/local/man which I think should be picked up by any man binary. You can also change it eg. make install mandir=/some/other/man. I'm also running macOS but using MacPorts.

Yeah. I find it strange as well. Played around a lot with paths and different instances of man today. I could potentially investigate a little bit more when I have some free time on my hands.

**Comment by mohd-akram** [comment]
Thanks for clarifying. I've updated the README with links to Pandoc and git-tools so people know how to install them. The git-tools README has instructions for all the popular package managers. And btw, you should not have needed to run `make` because it takes a long time. The repo already has all the files created so you can just install directly.

> Yeah. I find it strange as well. Played around a lot with paths and different instances of man today. I could potentially investigate a little bit more when I have some free time on my hands.

I would suggest running `which man` to see which version of man is being invoked, as well as running `man` with the `-d` flag to see what it's doing and where it's searching, eg. `man -d -S 3js eval`.

**Comment by mohd-akram** [comment]
Btw, if you want it to work without having to specify `-S 3js` on macOS, you have to add this to `/usr/local/etc/man.d/js.conf`:

```
MANSECT 3js
```

You will need the latest macOS for this to work, 14.4. Otherwise just use the `alias` trick I mention in the README which is convenient.

**Pull request closed without merging**